block-storage/src/main/scala/coop/rchain/blockstorage/BlockDagFileStorage.scala

```diff
@@ -1204,7 +1204,7 @@ object BlockDagFileStorage {
       _                      <- writeToFile[F](config.latestMessagesLogPath, latestMessagesData)
       _                      <- writeToFile[F](config.latestMessagesCrcPath, latestMessagesCrcBytes)
       blockMetadataCrc       = Crc32.empty[F]()
-      genesisByteString      = genesis.toByteString
+      genesisByteString      = BlockMetadata.fromBlock(genesis, invalid = false).toByteString
```

Comment:
Could you please explain why we are only going to store the genesis block meta data?

Reply:
That was a bug. We always store block metadata in DAG storage. Gensis is not an exception. This particular constructor (`createEmptyFromGenesis`) is only used in casper tests that is why  it didn't show itself before @lukasz-golebiewski wrote a test with restarting casper.

Reply:
Ah so we also separately store the genesis block in its own file. Makes sense.